core/src/main/java/org/apache/carbondata/core/keygenerator/columnar/impl/MultiDimKeyVarLengthEquiSplitGenerator.java

```diff
@@ -143,17 +142,18 @@ private void intialize() {
 
   private int[] convertToArray(List<Integer> list) {
```

Comment:
this is all depending on if we want to hold a primitive type array not not. these type of boxing/unboxing for primitive array type is annoying